state/state.go

```diff
@@ -1729,7 +1729,16 @@ func (st *State) AddRelation(eps ...Endpoint) (r *Relation, err error) {
 		return ops, nil
 	}
 	if err = st.run(buildTxn); err == nil {
-		return &Relation{st, *doc}, nil
+		return &Relation{st, *doc}, err
+	} else if errors.IsAlreadyExists(err) {
+		// If the relation already exists, we'll return it
```

Comment:
Where is it used? I cannot see any in state. Saving a facade RPC call is a noble goal, but doesn't mean the check needs to be done here.

Please at least document the behaviour in the doc comment.